pkg/plugin/register.go

```diff
@@ -24,21 +23,23 @@ type registerBody struct {
 	URL string `json:"url"`
 }
 
-func RegisterPlugin(listener net.Listener) {
+func RegisterPlugin(listener net.Listener) error {
 	if os.Getenv(StandaloneEnvVar) == "1" {
-		return
+		return nil
 	}
 
 	minimumNumberOfCLIArgument := 2
 
 	if len(os.Args) >= minimumNumberOfCLIArgument {
 		err := register(os.Args[1], listener.Addr())
 		if err != nil {
-			log.Panicln(err)
+			return err
 		}
 	} else {
 		panic("Usage: program must be started with a correlationID command line argument")
```

Comment:
I think we also want to return an error here ?

Reply:
no it's out of scope

Reply:
We agreed that it's out of scope and shouldn't really be an issue